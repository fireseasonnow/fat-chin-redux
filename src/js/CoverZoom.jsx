import React from 'react';
import styled from 'styled-components';

const CoverZoom = ({
    coverZoomHandler,
    coverUrl,
    name,
}) => (
    <Inner>
        <Close onClick={() => coverZoomHandler()} type='button' />
        <Img src={coverUrl} alt={`'${name}' album cover`} />
    </Inner>
);

export default CoverZoom;

const Inner = styled.div`
    display: grid;
    grid-template-columns: 1fr auto 1fr;
    height: 100%;
`;

const Close = styled.button`
    align-self: center;
    justify-self: end;
    background: none;
    color: #fff;
    border: 0;
    position: relative;
    left: -15px;
    width: 40px;
    height: 40px;
    padding-right: 28px;

    &:hover {
        cursor: pointer;
    }

    &:before {
        content: '';
        position: absolute;
        top: 20px;
        width: 21px;
        height: 1px;
        background-color: currentColor;
        transform: rotate(-45deg);
    }

    &:after {
        content: '';
        position: absolute;
        top: 20px;
        width: 21px;
        height: 1px;
        background-color: currentColor;
        transform: rotate(45deg);
    }
`;

const Img = styled.img`
    align-self: center;
`;
