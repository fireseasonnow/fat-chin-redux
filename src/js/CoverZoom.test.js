import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

import CoverZoom from './CoverZoom';

configure({ adapter: new Adapter() });

describe('<CoverZoom />', () => {
    let wrapper = {};

    beforeEach(() => {
        wrapper = shallow(<CoverZoom />);
    });

    it('should render without crashing', () => {
        expect(wrapper).toBeTruthy();
    });

    it('should render a cover zoom with name and coverUrl', () => {
        wrapper.setProps({
            name: 'Hello',
            coverUrl: 'url',
        });

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should fire coverZoomHandler when clicked on a button', () => {
        const coverZoomHandler = jest.fn();

        wrapper.setProps({
            coverZoomHandler,
        });

        wrapper.find('CoverZoom__Close').simulate('click');

        expect(coverZoomHandler).toHaveBeenCalled();
    });
});
