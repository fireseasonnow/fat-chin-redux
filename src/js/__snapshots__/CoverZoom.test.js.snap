// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoverZoom /> should render a cover zoom with name and coverUrl 1`] = `
<CoverZoom__Inner>
  <CoverZoom__Close
    onClick={[Function]}
    type="button"
  />
  <CoverZoom__Img
    alt="'Hello' album cover"
    src="url"
  />
</CoverZoom__Inner>
`;
